src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -640,18 +641,18 @@ internal string GetServiceName(Tracer tracer, string serviceName)
             return ServiceNameMappings.GetServiceName(tracer.DefaultServiceName, serviceName);
         }
 
-        private static HashSet<PropagatorType> GetPropagators(IConfigurationSource source)
+        private static HashSet<string> GetPropagators(IConfigurationSource source)
         {
-            var propagators = source.GetTypedValues<PropagatorType>(ConfigurationKeys.Propagators);
+            var propagators = source.GetStrings(ConfigurationKeys.Propagators);
 
             if (!propagators.Any())
```

Comment:
This will cause multiple enumeration (at `.Any()` and then at `HashSet` creation). I know there will always be an array underneath but I would add `.ToArray()` in that case just for the sake of it. We won't risk it breaking if the implementation changes.

Reply:
Originally it was intentional due perf and allocation benefits (Enumerate first + enumerate all vs allocate new list / array). @pellared any ideas which way is more preferable?

Reply:
I would not worry much about the perf for such cases since it is executed once per application lifetime. Of course, I'm not saying to ignore performance and to not do easy/obvious savings but the impact here will be very minimal anyway.